src/go/k8s/pkg/console/store.go

```diff
@@ -0,0 +1,107 @@
+package console
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+)
+
+// By default, Console can only be created with the same namespace as the referenced Cluster.
+// This is because Secrets, which are namespaced resource, cannot be mounted in Deployment (e.g. Schema Registry TLS).
+// Store enables creating Console in different namespace from the Cluster by saving a copy of the Secret in cache.
```

Comment:
I am kind of curious why you chose this path why not just copy secret to other namespace when needed? 🤔 we do that for some secrets as well

Reply:
Right now we are just copying Secrets, in the future we might sync other namespaced resources (e.g. configmaps) I feel that we need an abstraction to sync these easily. Although this package is not perfect but it starts the idea. This also just:
- Copies the secret
- Sync them (provided you watch the right events in `SetupWithManager`)
- Generic (can sync any resource)
- Cache them in a store which you can access the freshest value anywhere in the controller